ssl4rs/models/classif/base.py

```diff
@@ -367,3 +368,138 @@ def _render_and_log_samples(
             self._log_rendered_image(output_image, key=f"{loop_type}/{sample_id}")
             outputs.append(output_image)
         return outputs
+
+#todo: refactor loss to relevant file
+class MaskedMSELoss(nn.Module):
+    def __init__(self, ignore_index=-1):
+        super(MaskedMSELoss, self).__init__()
+        self.ignore_index = ignore_index
+        self.mse_loss = torch.nn.MSELoss()
+
+    def forward(self, predictions, targets):
+        mask = targets != self.ignore_index
+        valid_predictions = predictions[mask]
+        valid_targets = targets[mask]
+
+        loss = self.mse_loss(valid_predictions, valid_targets)
+        return loss
+
+#todo: refactor metric to relevant file
+class MaskedMeanSquaredError(torchmetrics.Metric):
+    def __init__(self, ignore_index=-1, dist_sync_on_step=False):
+        super(MaskedMeanSquaredError, self).__init__(dist_sync_on_step=dist_sync_on_step)
+        self.ignore_index = ignore_index
+        self.mse_metric = torchmetrics.MeanSquaredError()
+
+    def update(self, predictions: torch.Tensor, targets: torch.Tensor):
+        mask = targets != self.ignore_index
+
+        valid_predictions = predictions[mask]
+        valid_targets = targets[mask]
+        self.mse_metric.update(valid_predictions, valid_targets)
+
+    def compute(self):
+        return self.mse_metric.compute()
+
+    def reset(self):
+        self.mse_metric.reset()
+
+
+class SegmenterBoundaryDistance(GenericSegmenter):
```

Comment:
this class should be in a new (seperate) module, currently it's hidden in a place we don't expect to find it